### PR TITLE
stm32: update metapac

### PR DESCRIPTION
**Pull request by xoviat**

### CODE REVIEW

File: embassy-stm32/Cargo.toml [hunk 1]
```diff
@@ -57,7 +57,7 @@ sdio-host = "0.5.0"
 embedded-sdmmc = { git = "https://github.com/embassy-rs/embedded-sdmmc-rs", rev = "a4f293d3a6f72158385f79c98634cb8a14d0d2fc", optional = true }
 critical-section = "1.1"
 atomic-polyfill = "1.0.1"
-stm32-metapac = "12"
+stm32-metapac = "13"
```
Comment (Dirbaio) [inline embassy-stm32/Cargo.toml:60]: need to update in `build-dependencies` too

File: embassy-stm32/src/dac/mod.rs [hunk 2]
```diff
@@ -51,7 +51,10 @@ impl Ch1Trigger {
     fn tsel(&self) -> dac::vals::Tsel1 {
         match self {
             Ch1Trigger::Tim6 => dac::vals::Tsel1::TIM6_TRGO,
+            #[cfg(not(dac_v3))]
             Ch1Trigger::Tim3 => dac::vals::Tsel1::TIM3_TRGO,
+            #[cfg(dac_v3)]
+            Ch1Trigger::Tim3 => dac::vals::Tsel1::TIM1_TRGO,
```
Comment (Dirbaio) [inline embassy-stm32/src/dac/mod.rs:57]: this seems wrong

Comment (xoviat) [reply embassy-stm32/src/dac/mod.rs:57]: @JuliDi will fix it in the next pr

Comment (JuliDi) [reply embassy-stm32/src/dac/mod.rs:57]: This should do:
```rust
#[derive(Debug, Copy, Clone, Eq, PartialEq)]
#[cfg_attr(feature = "defmt", derive(defmt::Format))]
/// Trigger sources for CH1
pub enum Ch1Trigger {
    #[cfg(dac_v3)]
    Tim1,
    Tim2,
    #[cfg(dac_v2)]
    Tim3,
    #[cfg(dac_v3)]
    Tim4,
    #[cfg(dac_v3)]
    Tim5,
    Tim6,
    Tim7,
    #[cfg(dac_v3)]
    Tim8,
    Tim15,
    #[cfg(dac_v3)]
    Hrtim1Dactrg1,
    #[cfg(dac_v3)]
    Hrtim1Dactrg2,
    #[cfg(dac_v3)]
    Lptim1,
    #[cfg(dac_v3)]
    Lptim2,
    #[cfg(dac_v3)]
    Lptim3,
    Exti9,
    Software,
}

impl Ch1Trigger {
    fn tsel(&self) -> dac::vals::Tsel1 {
        match self {
            #[cfg(dac_v3)]
            Ch1Trigger::Tim1 => dac::vals::Tsel1::TIM1_TRGO,
            Ch1Trigger::Tim2 => dac::vals::Tsel1::TIM2_TRGO,
            #[cfg(dac_v2)]
            Ch1Trigger::Tim3 => dac::vals::Tsel1::TIM3_TRGO,
            #[cfg(dac_v3)]
            Ch1Trigger::Tim4 => dac::vals::Tsel1::TIM4_TRGO,
            #[cfg(dac_v3)]
            Ch1Trigger::Tim5 => dac::vals::Tsel1::TIM5_TRGO,
            Ch1Trigger::Tim6 => dac::vals::Tsel1::TIM6_TRGO,
            Ch1Trigger::Tim7 => dac::vals::Tsel1::TIM7_TRGO,
            #[cfg(dac_v3)]
            Ch1Trigger::Tim8 => dac::vals::Tsel1::TIM8_TRGO,
            Ch1Trigger::Tim15 => dac::vals::Tsel1::TIM15_TRGO,
            #[cfg(dac_v3)]
            Ch1Trigger::Hrtim1Dactrg1 => dac::vals::Tsel1::HRTIM1_DACTRG1,
            #[cfg(dac_v3)]
            Ch1Trigger::Hrtim1Dactrg2 => dac::vals::Tsel1::HRTIM1_DACTRG2,
            #[cfg(dac_v3)]
            Ch1Trigger::Lptim1 => dac::vals::Tsel1::LPTIM1_OUT,
            #[cfg(dac_v3)]
            Ch1Trigger::Lptim2 => dac::vals::Tsel1::LPTIM2_OUT,
            #[cfg(dac_v3)]
            Ch1Trigger::Lptim3 => dac::vals::Tsel1::LPTIM3_OUT,
            Ch1Trigger::Exti9 => dac::vals::Tsel1::EXTI9,
            Ch1Trigger::Software => dac::vals::Tsel1::SOFTWARE,
        }
    }
}
```

Comment (JuliDi) [reply embassy-stm32/src/dac/mod.rs:57]: But I'll add it to the next PR. And @Dirbaio is right, it should not be v3 for the `Tim3` but v2

**Review by Dirbaio — APPROVED** [review]
okay, let's fix later

**Pull request merged**